src/entry-client.js

```diff
@@ -5,6 +5,54 @@ const shouldIgnoreRouteUpdate = (c, args) => (
     c.shouldIgnoreRouteUpdate(args) === true
 );
 
+// To be toggled on via client options if desired
+let enablePerfMarks = false;
+
+const PERF_PREFIX = 'urbnperf';
+const perfEnabled = () => enablePerfMarks && window.performance !== null;
+
+// Look up the current perf mark of the format urbnperf|*|start
+const getCurrentPerfMark = () => window.performance.getEntriesByType('mark')
+    .find(m => m.name.startsWith(PERF_PREFIX) && m.name.endsWith('start'));
+
+function perfInit(to, from) {
+    if (!perfEnabled()) {
+        return;
+    }
+
+    // Always clear any prior measurements before starting a new one
+    window.performance.getEntriesByType('mark')
+        .filter(m => m.name.startsWith(PERF_PREFIX))
+        .forEach(m => window.performance.clearMarks(m.name));
+
+    window.performance.getEntriesByType('measure')
+        .filter(m => m.name.startsWith(PERF_PREFIX))
+        .forEach(m => window.performance.clearMeasures(m.name));
+
+    // Start a new routing operation with a mark such as:
+    //   urbnperf|Homepage->Catch-All|start
+    window.performance.mark(`${PERF_PREFIX}|${from.name}->${to.name}|start`);
+}
+
+// Issue a performance.measure call for the given name using the most recent
+// 'start' mark
+export function perfMeasure(name) {
```

Comment:
Keep this as simple as possible to reduce app code complexity - and always base measures off the most recent mark.  We make the assumption that there is only ever one active mark/measure cycle, which is ensured by clearing any prior marks/measures in `perfInit`.  That way the app never needs to figure out the `to`/`from` values